parse/src/main/java/com/parse/ParseObjectStoreMigrator.java

```diff
@@ -0,0 +1,71 @@
+package com.parse;
+
+import com.parse.boltsinternal.Continuation;
+import com.parse.boltsinternal.Task;
+
+import java.util.Arrays;
+
+public class ParseObjectStoreMigrator<T extends ParseObject> implements ParseObjectStore<T> {
```

Comment:
You could add some Javadoc with comments. 

Does this migrator needs to be here forever or just for about some migration period?

Reply:
> Wow @DrMimik! Many thanks for this contribution. It's highly appreciated! Looks very clean and nice!
>
> I have the feeling that this is something working for a long time somewhere else, correct me if I'm wrong. I'm guessing that because of the copyright comments.
> 
> I left just few questions and small javadoc suggestions. I'm wondering @mtrezza if the migration mechanism should be here forever or there is some policy about that. But if @DrMimik say that already migrated stores are covered and there's no performance issue maybe can stay for a while.

Thank you guys for creating this awesome platform, I actuallty just copied the `ParseObjectStore`'s logic for creating a new ObjectStore and used Jetpack's crypto library.

Since Robolectric doesn't mock Android's `java.security.KeyStore` I copied a couple files from this [repo](https://github.com/appmattus/layercache/tree/main/testutils/src/main/kotlin/com/appmattus/layercache/keystore) (with their copyright XD ) to create tests using Robolectric.

Reply:
> We don't have a specific policy for the duration of providing migration mechanisms. It depends on the type of change. In this case I'd see the mechanism staying for several years, so indefinite at this point. The reason is that it can be a years long process to migrate clientes once they are released to end-users.

I think the migrator should be available for new users until they migrate, or remove it on a major release.